programmers/크레인-인형뽑기-게임.js

```diff
@@ -0,0 +1,37 @@
+function pullDoll(board, move) {
+  const dollIndex = board.findIndex((column) => column[move] !== 0);
+  if (dollIndex === -1) return -1;
+  const doll = board[dollIndex][move];
+  board[dollIndex][move] = 0;
+  return doll;
+}
+
+function solution(board, moves) {
+  moves = moves.map((move) => move - 1);
+  let basket = [];
+  let poppedTime = 0;
+  basket.pop = function () {
+    poppedTime += 2;
+    return Array.prototype.pop.apply(this, arguments);
+  }
+  moves.forEach((move) => {
+    const doll = pullDoll(board, move);
+    if (doll === -1) return;
+    if (doll === basket[basket.length - 1]) basket.pop();
+    else basket.push(doll);
+  });
+  return poppedTime;
+}
```

Comment:
배열에 메서드를 추가할 생각을 하다니 굳!
function()과 에로우 펑션의 this 차이를 잘 알고 있는 것 같네요.
실제로 this를 바인딩하여 사용하다니 놀랍구요.
나중에 코드에 대해서 설명 좀 부탁해요.

this를 바인딩하여 각 스코프를 기억하게 한 것같은데 정확한 동작 흐름에 대해 궁금하네요.
개발은 어느 정도할지 또 궁금해지네요 :D

줄바꿈만 조금 해줬습니다.

```suggestion
function solution(board, moves) {
  moves = moves.map((move) => move - 1);
  
  let basket = [];
  let poppedTime = 0;
  basket.pop = function () {
    poppedTime += 2;
    return Array.prototype.pop.apply(this, arguments);
  }
  
  moves.forEach((move) => {
    const doll = pullDoll(board, move);
    if (doll === -1) return;
    if (doll === basket[basket.length - 1]) basket.pop();
    else basket.push(doll);
  });
  return poppedTime;
}
```